### PR TITLE
Mitaka release

### PR DESCRIPTION
**Pull request by ramereth**
This is a large pull-request unfortunately and would like to have the review started on this.

Summary of changes

* Refactor of code to use newer Mitaka cookbooks refactor
* Split out individual OS services into their own recipes
* Added more integration tests
* Added ChefSpec tests
* Added support for Neutron using Linuxbridge support

Some things that are still a work in progress:
* [x] Create special qemu-ev package for x86 which is linked against GlusterFS 3.4
* [x] Remove unused databags
* [x] Ensure Linuxbridge ``local_ip`` is on same network for compute/controller nodes
* [x] Verify chef provisioning env is working
* [x] Clean out any extra TK suites that aren't needed anymore
* [x] Upstream openstack-dashboard patches to [#1](https://review.openstack.org/#/c/344494/)
* [x] Upstream openstack-dashboard [#2](https://review.openstack.org/#/c/346899/)
* [ ] Upstream openstack-dashboard [#3](https://review.openstack.org/#/c/349772/)

### CODE REVIEW

**Comment by armiller** [comment]
@ramereth to make this easier to track I'm going to :+1: individual files

**Comment by ramereth** [comment]
@armiller sounds good to me

**Comment by armiller** [comment]
Files to review:
- [ ] .chef/knife.rb
- [ ] .gitignore
- [ ] .kitchen.yml
- [ ] .rubocop.yml
- [ ] Berksfile
- [ ] Rakefile
- [ ] attributes/default.rb
- [ ] metadata.rb
- [ ] recipes/_fedora.rb
- [ ] recipes/block_storage.rb
- [ ] recipes/block_storage_controller.rb
- [ ] recipes/cinder.rb
- [ ] recipes/compute.rb
- [ ] recipes/compute_controller.rb
- [ ] recipes/controller.rb
- [ ] recipes/dashboard.rb
- [ ] recipes/default.rb
- [ ] recipes/identity.rb
- [ ] recipes/image.rb
- [ ] recipes/linuxbridge.rb
- [ ] recipes/network.rb
- [ ] recipes/novnc.rb
- [ ] recipes/ops_database.rb
- [ ] recipes/ops_messaging.rb
- [ ] recipes/telemetry.rb
- [ ] spec/block_storage_controller_spec.rb
- [ ] spec/block_storage_spec.rb
- [ ] spec/compute_controller_spec.rb
- [ ] spec/compute_spec.rb
- [ ] spec/controller_spec.rb
- [ ] spec/dashboard_spec.rb
- [ ] spec/default_spec.rb
- [ ] spec/identity_spec.rb
- [ ] spec/image_spec.rb
- [ ] spec/linuxbridge_spec.rb
- [ ] spec/network_spec.rb
- [ ] spec/ops_messaging_spec.rb
- [ ] spec/spec_helper.rb
- [ ] spec/telemetry_spec.rb
- [ ] templates/default/novncproxy.erb
- [ ] test/chef-config/knife.rb
- [ ] test/cookbooks/openstack_test/metadata.rb
- [ ] test/cookbooks/openstack_test/recipes/compute.rb
- [ ] test/cookbooks/openstack_test/recipes/default.rb
- [ ] test/cookbooks/openstack_test/recipes/gluster.rb
- [ ] test/integration/block_storage/serverspec/block_storage_spec.rb
- [ ] test/integration/block_storage_controller/serverspec/block_storage_controller_spec.rb
- [ ] test/integration/chef-provisioning/compute.rb
- [ ] test/integration/chef-provisioning/controller.rb
- [ ] test/integration/chef-provisioning/controller_compute.rb
- [ ] test/integration/chef-provisioning/vagrant_linux.rb
- [ ] test/integration/cinder/serverspec/server_spec.rb
- [ ] test/integration/compute/serverspec/compute_spec.rb
- [ ] test/integration/compute_controller/serverspec/compute_controller_spec.rb
- [ ] test/integration/controller/serverspec/controller_spec.rb
- [ ] test/integration/dashboard/serverspec/dashboard_spec.rb
- [ ] test/integration/default/data_bags/certificates/novnc.json
- [ ] test/integration/default/data_bags/certificates/wildcard.json
- [ ] test/integration/default/data_bags/certs/horizon.key.json
- [ ] test/integration/default/data_bags/certs/horizon.pem.json
- [ ] test/integration/default/data_bags/x86_db_passwords/gnocchi.json
- [ ] test/integration/default/data_bags/x86_db_passwords/ironic.json
- [ ] test/integration/default/data_bags/x86_db_passwords/mysqlroot.json
- [ ] test/integration/default/data_bags/x86_db_passwords/nova_api.json
- [ ] test/integration/default/data_bags/x86_secrets/dispersion_auth_key.json
- [ ] test/integration/default/data_bags/x86_secrets/dispersion_auth_user.json
- [ ] test/integration/default/data_bags/x86_secrets/openstack_metering_secret.json
- [ ] test/integration/default/data_bags/x86_secrets/orchestration_auth_encryption_key.json
- [ ] test/integration/default/data_bags/x86_secrets/swift_authkey.json
- [ ] test/integration/default/data_bags/x86_secrets/swift_hash_path_prefix.json
- [ ] test/integration/default/data_bags/x86_secrets/swift_hash_path_suffix.json
- [ ] test/integration/default/data_bags/x86_service_passwords/openstack-bare-metal.json
- [ ] test/integration/default/data_bags/x86_service_passwords/openstack-object-storage.json
- [ ] test/integration/default/data_bags/x86_service_passwords/openstack-telemetry-metric.json
- [ ] test/integration/default/data_bags/x86_service_passwords/openstack-telemetry.json
- [ ] test/integration/default/data_bags/x86_user_passwords/mysqlroot.json
- [ ] test/integration/default/serverspec/default_spec.rb
- [ ] test/integration/identity/serverspec/identity_spec.rb
- [ ] test/integration/image/serverspec/image_spec.rb
- [ ] test/integration/linuxbridge/serverspec/linuxbridge_spec.rb
- [ ] test/integration/network/serverspec/network_spec.rb
- [ ] test/integration/ops_messaging/serverspec/ops_messaging_spec.rb
- [ ] test/integration/roles/openstack_cinder.json
- [ ] test/integration/roles/openstack_provisioning.json
- [ ] test/integration/roles/openstack_tk.json
- [ ] test/integration/roles/openstack_vagrant.json
- [ ] test/integration/roles/vagrant_provisioning.json
- [ ] test/integration/telemetry/serverspec/telemetry_spec.rb

File: Berksfile [hunk 1]
```diff
@@ -2,39 +2,45 @@ source 'https://supermarket.chef.io'
...
 cookbook 'statsd', github: 'att-cloud/cookbook-statsd'
-cookbook 'yum', '>= 3.1.4'
-cookbook 'yum-epel', '>= 0.3.4'
-cookbook 'yum-fedora'
+cookbook 'yum-qemu-ev', git: 'git@github.com:osuosl-cookbooks/yum-qemu-ev.git'
+
+# WIP patches
+%w(
+  dashboard
+).each do |cb|
```
Comment (armiller) [inline Berksfile]: Is this loop still needed if there is only one element?

File: recipes/block_storage.rb [hunk 2]
```diff
@@ -16,26 +16,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-# this is required because of the fedora deps. Will be fixed once its moved into
-# a _common recipe.
 iscsi_hosts = ['127.0.0.1']
 iscsi_hosts += node['osl-openstack']['cinder']['iscsi_ips']
 iscsi_role = node['osl-openstack']['cinder']['iscsi_role']
-unless iscsi_role.nil? && Chef::Config[:solo]
+unless (iscsi_role.nil? && Chef::Config[:solo]) || !defined?(ChefSpec)
```
Comment (armiller) [inline recipes/block_storage.rb]: I would rewrite this to not use unless. That is really hard logic to follow.

File: recipes/compute.rb [hunk 3]
```diff
@@ -16,57 +16,39 @@
...
+    not_if '/sbin/ppc64_cpu --smt 2>&1 | ' \
+      'grep -E \'SMT is off|Machine is not SMT capable\''
   end
 end
 
+include_recipe 'osl-openstack::linuxbridge'
 include_recipe 'openstack-compute::compute'
-include_recipe 'openstack-compute::api-metadata'
+include_recipe 'openstack-telemetry::agent-compute'
+# include_recipe 'openstack-bare-metal::conductor'
```
Comment (armiller) [inline recipes/compute.rb]: Why is this still commented out?

File: recipes/compute_controller.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+certificate_manage 'novnc' do
+  cert_path ssl_dir
+  cert_file novnc['cert_file']
+  key_file  novnc['key_file']
+  chain_file 'novnc-bundle.crt'
+  nginx_cert true
+  owner node['openstack']['compute']['user']
+  group node['openstack']['compute']['group']
+  notifies :restart, resources(proxy_service)
```
Comment (armiller) [inline recipes/compute_controller.rb]: What is `resources` doing here? The template below is not using it.

File: recipes/controller.rb [hunk 5]
```diff
@@ -16,41 +16,25 @@
...
+include_recipe 'osl-openstack::compute_controller'
+include_recipe 'osl-openstack::block_storage_controller'
+include_recipe 'osl-openstack::telemetry'
+# include_recipe 'openstack-bare-metal::api'
+# include_recipe 'openstack-bare-metal::identity_registration'
+# include_recipe 'openstack-orchestration::engine'
+# include_recipe 'openstack-orchestration::api'
+# include_recipe 'openstack-orchestration::api-cfn'
+# include_recipe 'openstack-orchestration::api-cloudwatch'
+# include_recipe 'openstack-orchestration::identity_registration'
```
Comment (armiller) [inline recipes/controller.rb]: Can these be removed?

File: recipes/dashboard.rb [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+              'openstack_dashboard',
+              'local',
+              '.secret_key_store')
+
+file secret_lock_file do
+  owner 'root'
+  group node['openstack']['dashboard']['horizon_user']
+  mode 0660
+  subscribes :create, 'service[apache2]', :immediately
+  only_if { File.exist?(secret_lock_file) }
```
Comment (armiller) [inline recipes/dashboard.rb]: Is there a reason `File` is not namespaced?

File: recipes/default.rb [hunk 7]
```diff
@@ -16,68 +16,148 @@
...
+end
+node.default['openstack']['release'] = 'mitaka'
 node.default['openstack']['secret']['key_path'] =
   '/etc/chef/encrypted_data_bag_secret'
-node.default['openstack']['sysctl']['net.ipv4.conf.all.rp_filter'] = 0
-node.default['openstack']['sysctl']['net.ipv4.conf.default.rp_filter'] = 0
-node.default['openstack']['sysctl']['net.ipv4.ip_forward'] = 1
+# node.default['openstack']['sysctl']['net.ipv4.conf.all.rp_filter'] = 0
+# node.default['openstack']['sysctl']['net.ipv4.conf.default.rp_filter'] = 0
+# node.default['openstack']['sysctl']['net.ipv4.ip_forward'] = 1
```
Comment (armiller) [inline recipes/default.rb]: Do these still need to be commented out?

File: Berksfile (hunk 1 above)
Comment (ramereth) [inline Berksfile]: It's nice to have if I have more than one cookbook I need to use in WIP. This should go away by the time we need to merge this cookbook (I hope).

File: recipes/compute.rb (hunk 3 above)
Comment (ramereth) [inline recipes/compute.rb]: This is an artifact from prior testing. It should be removed.

File: recipes/controller.rb (hunk 5 above)
Comment (ramereth) [inline recipes/controller.rb]: Yes

File: recipes/compute_controller.rb (hunk 4 above)
Comment (ramereth) [inline recipes/compute_controller.rb]: I honestly don't remember... The intention is that when you update the cert it needs to restart the proxy service.

File: recipes/dashboard.rb (hunk 6 above)
Comment (ramereth) [inline recipes/dashboard.rb]: Nope, forgot to do that.

File: recipes/default.rb (hunk 7 above)
Comment (ramereth) [inline recipes/default.rb]: Yes, I think so as this I think this gets set in the upstream cookbooks. I'll verify that and remove if it needs to be.

File: spec/block_storage_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,94 @@
+require_relative 'spec_helper'
+require 'chef/application'
+
+describe 'osl-openstack::block_storage_controller',
+         block_storage_controller: true do
+  let(:runner) do
+    ChefSpec::SoloRunner.new(REDHAT_OPTS) do |node|
```
Comment (armiller) [inline spec/block_storage_controller_spec.rb]: Is there a reason this is redhat and not centos?

**Comment by armiller** [comment]
`mon-centos-72` serverspec test is missing quotes around check_load.cfg

**Comment by armiller** [comment]
@ramereth code wise it gets a +1 from me. Is it possible for me to tests this using chef-provision?

**Comment by ramereth** [comment]
@armiller I'm working on fixing that still

**Comment by ramereth** [comment]
@osuosl-cookbooks/chefs this is ready for a final review before merging

**Comment by armiller** [comment]
👍

**Comment by ramereth** [comment]
!bump major

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a major-level version bump to v2.0.0.  Have a nice day!

**Pull request merged**